test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EnumFilterFunctionalTests.cs

```diff
@@ -515,28 +520,245 @@ public void ParseFilterEnumTypesWrongCast3()
+            string filterQuery = $"{integralValue} in Colors";
+
+            // Act
+            Action action = () => ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+
+            // Assert
+            action.Throws<ODataException>(Strings.Binder_IsNotValidEnumConstant("-20"));
+        }
+
+        [Fact]
+        public void ParseFilterWithInOperatorWithEnumsMemberNameWithoutSingleQuotes_ThrowsNullReferenceException()
+        {
+            // Arrange
+            string filterQuery = "Red in Colors";
+
+            string erroMessage = "Object reference not set to an instance of an object.";
+
+            // Act
+            Action action = () => ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+
+            // Assert
+            action.Throws<NullReferenceException>(erroMessage);
+        }
+
+        [Theory]
+        [InlineData("'Yellow'")]
+        [InlineData("'Teal'")]
+        [InlineData("NS.Color'Yellow'")]
+        [InlineData("NS.Color'Teal'")]
+        public void ParseFilterWithInOperatorWithEnumsInvalidMemberNames_ThrowsIsNotValidEnumConstantException(string memberName)
+        {
+            // Arrange
+            string filterQuery = $"{memberName} in Colors";
+
+            string expectedExceptionParameter = memberName.StartsWith("'") ? memberName.Trim('\'') : memberName; // Trim('\'') method removes any single quotes from the start and end of the string
+
+            // Act
+            Action action = () => ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+
+            // Assert
+            action.Throws<ODataException>(Strings.Binder_IsNotValidEnumConstant(expectedExceptionParameter));
+        }
+
+        private T GetIEdmType<T>(string typeName) where T : IEdmType
         {
-            return (IEdmStructuralProperty)((IEdmStructuredType)model
-                .FindType("NS.MyEntityType"))
-                .FindProperty("Color");
+            return (T)this.userModel.FindType(typeName);
         }
 
-        private IEdmEnumType GetColorType(IEdmModel model)
+        private T GetIEdmType<T>(IEdmModel model, string typeName) where T : IEdmType
         {
-            return (IEdmEnumType)model.FindType("NS.Color");
+            return (T)model.FindType(typeName);
         }
 
-        private IEdmStructuralProperty GetColorFlagsProp(IEdmModel model)
+        private IEdmStructuralProperty GetIEdmProperty(string propertyName)
         {
-            return (IEdmStructuralProperty)((IEdmStructuredType)model
-                .FindType("NS.MyEntityType"))
-                .FindProperty("ColorFlags");
+            return (IEdmStructuralProperty)this.GetIEdmType<IEdmStructuredType>("NS.MyEntityType")
+                .FindProperty(propertyName);
         }
 
-        private IEdmEnumType GetColorFlagsType(IEdmModel model)
+        private IEdmStructuralProperty GetIEdmProperty(string entityName, string propertyName)

```

Comment:
Could also use an optional param with a default value instead of overloads. I think that would make it easier for the function caller to see what the default is?